test/qa/lib/meshnet.py

```diff
@@ -20,27 +20,141 @@
+        self.os = os
+        self.distribution = distribution
+        self.allow_incoming_traffic = self._convert_to_bool(allow_incoming_traffic)
+        self.allow_routing = self._convert_to_bool(allow_routing)
+        self.allow_lan_access = self._convert_to_bool(allow_lan_access)
+        self.allow_sending_files = self._convert_to_bool(allow_sending_files)
+        self.allows_incoming_traffic = self._convert_to_bool(allows_incoming_traffic)
+        self.allows_routing = self._convert_to_bool(allows_routing)
+        self.allows_lan_access = self._convert_to_bool(allows_lan_access)
+        self.allows_sending_files = self._convert_to_bool(allows_sending_files)
+        self.accept_fileshare_automatically = self._convert_to_bool(accept_fileshare_automatically)
+
+    def _convert_to_bool(self, value):
+        return value.lower() == "enabled" if value is not None else None
+
+
 class PeerName(Enum):
     Hostname = 0
     Ip = 1
     Pubkey = 2
 
 
+class PeerList:
+    def __init__(self):
+        self.this_device: list[Peer] = []
+        self.internal_peers: list[Peer] = []
+        self.external_peers: list[Peer] = []
+
+    def _str_to_dictionary(self, data: str):
+        new_dictionary = {a.strip().lower(): b.strip()
+                for a, b in (element.split(':')
+                                for element in
+                                filter(lambda line: len(line.split(':')) == 2, data.split('\n')))}
+
+        return new_dictionary
+
+    def _add_peer(self, to_peer_list: list[Peer], peer_data: str):
+        peer_data_dictionary = self._str_to_dictionary(peer_data)
+
+        # This device case
+        peer = Peer(
+                hostname = peer_data_dictionary['hostname'],
+                nickname = peer_data_dictionary["nickname"],
+                ip = peer_data_dictionary["ip"],
+                public_key = peer_data_dictionary["public key"],
+                os = peer_data_dictionary["os"],
+                distribution = peer_data_dictionary["distribution"],
+            )
+
+        # Internal, external peer cases
+        if "Status" in peer_data_dictionary:
+            peer.status = peer_data_dictionary["status"]
+            peer.allow_incoming_traffic = peer_data_dictionary["allow incoming traffic"]
+            peer.allow_routing = peer_data_dictionary["allow routing"]
+            peer.allow_lan_access = peer_data_dictionary["allow local network access"]
+            peer.allow_sending_files = peer_data_dictionary["allow sending files"]
+            peer.allows_incoming_traffic = peer_data_dictionary["allows incoming traffic"]
+            peer.allows_routing = peer_data_dictionary["allows routing"]
+            peer.allows_lan_access = peer_data_dictionary["allows local network access"]
+            peer.allows_sending_files = peer_data_dictionary["allows sending files"]
+            peer.accept_fileshare_automatically = peer_data_dictionary["accept fileshare automatically"]
```

Comment:
Move all of this and also `_str_to_dictionary` to `Peer` class. You can create `Peer.from_str()` classmethod which will be used as an alternative constructor for `Peer` objects:

``` Python
class Peer:
    def __init__(
            self,
            hostname: str,
            nickname: str,
            ip: str,
            public_key: str,
            os: str,
            distribution: str,
            status: str = None,
            ):
        self.hostname = hostname
        self.nickname = nickname
        self.ip = ip
        self.public_key = public_key
        self.os = os
        self.distribution = distribution
        self.status = status
        self.allow_incoming_traffic = None
        self.allow_routing = None
        self.allow_lan_access = None
        self.allow_sending_files = None
        self.allows_incoming_traffic = None
        self.allows_routing = None
        self.allows_lan_access = None
        self.allows_sending_files = None
        self.accept_fileshare_automatically = None

    @classmethod
    def from_str(cls, data: str):
        "Place all the string parsing logic here"
        peer = cls(
                hostname = peer_data_dictionary['hostname'],
                nickname = peer_data_dictionary["nickname"],
                ip = peer_data_dictionary["ip"],
                public_key = peer_data_dictionary["public key"],
                os = peer_data_dictionary["os"],
                distribution = peer_data_dictionary["distribution"],
            )

        # Internal, external peer cases
        if "Status" in peer_data_dictionary:
            peer.status = peer_data_dictionary["status"]
            peer.allow_incoming_traffic = self._convert_to_bool(peer_data_dictionary["allow incoming traffic"])
            peer.allow_routing = self._convert_to_bool(peer_data_dictionary["allow routing"])
            peer.allow_lan_access = self._convert_to_bool(peer_data_dictionary["allow local network access"])
            peer.allow_sending_files = self._convert_to_bool(peer_data_dictionary["allow sending files"])
            peer.allows_incoming_traffic = self._convert_to_bool(peer_data_dictionary["allows incoming traffic"])
            peer.allows_routing = self._convert_to_bool(peer_data_dictionary["allows routing"])
            peer.allows_lan_access = self._convert_to_bool(peer_data_dictionary["allows local network access"])
            peer.allows_sending_files = self._convert_to_bool(peer_data_dictionary["allows sending files"])
            peer.accept_fileshare_automatically = self._convert_to_bool(peer_data_dictionary["accept fileshare automatically"])

        return peer
```
And then in `PeerList` just use for example:
``` Python
   def set_this_device(self, peer_data: str):
        self.this_device.append(Peer.from_str(peer_data))
```